src/views/lighttable.c

```diff
@@ -1183,66 +1183,58 @@ int expose_full_preview(dt_view_t *self, cairo_t *cr, int32_t width, int32_t hei
   if(lib->track < -2) offset--;
   lib->track = 0;
 
-  if(offset)
-  {
-    /* If more than one image is selected, iterate over these. */
-    /* If only one image is selected, scroll through all known images. */
-
-    int sel_img_count = 0;
-    {
-      sqlite3_stmt *stmt;
-      DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "select COUNT(*) from selected_images", -1,
-                                  &stmt, NULL);
-      if(sqlite3_step(stmt) == SQLITE_ROW)
-      {
-        sel_img_count = sqlite3_column_int(stmt, 0);
-      }
-      sqlite3_finalize(stmt);
-    }
-
-    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, lib->full_preview_id, 'r');
-
-    /* Build outer select criteria */
-    gchar *filter_criteria
-        = g_strdup_printf("INNER JOIN memory.collected_images AS col ON s1.imgid=col.imgid WHERE col.rowid "
-                          "%s %d ORDER BY rowid %s LIMIT 1",
-                          (offset > 0) ? ">" : "<", lib->full_preview_rowid, (offset > 0) ? "ASC" : "DESC");
-    dt_image_cache_read_release(darktable.image_cache, img);
+  /* If more than one image is selected, iterate over these. */
+  /* If only one image is selected, scroll through all known images. */
 
+  int sel_img_count = 0;
+  {
     sqlite3_stmt *stmt;
-    gchar *stmt_string = NULL;
-    if(sel_img_count > 1)
-    {
-      stmt_string = g_strdup_printf(
-          "SELECT col.imgid AS id, col.rowid FROM (SELECT imgid FROM selected_images) AS s1 %s",
-          filter_criteria);
-
-      DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), stmt_string, -1, &stmt, NULL);
-    }
-    else
+    DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "select COUNT(*) from selected_images", -1,
+                                &stmt, NULL);
+    if(sqlite3_step(stmt) == SQLITE_ROW)
     {
-      /* We need to augment the current main query a bit to fetch the
-       * row we need. */
-      const char *main_query = sqlite3_sql(lib->statements.main_query);
-      stmt_string = g_strdup_printf("SELECT col.imgid AS id, col.rowid FROM (%s) AS s1 %s", main_query,
-                                    filter_criteria);
-
-      DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), stmt_string, -1, &stmt, NULL);
-      DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, 0);
-      DT_DEBUG_SQLITE3_BIND_INT(stmt, 2, -1);
+      sel_img_count = sqlite3_column_int(stmt, 0);
     }
-    g_free(stmt_string);
-    g_free(filter_criteria);
+    sqlite3_finalize(stmt);
+  }
 
-    if(sqlite3_step(stmt) == SQLITE_ROW)
+  sqlite3_stmt *stmt;
+  gchar *stmt_string = NULL;
+  /* How many images to preload in advance. Three is just a random preference. */
+  int preload_num = 3;
```

Comment:
I think, this should be a config option.
See #975 for an example on how to add one...